app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt

```diff
@@ -58,6 +58,38 @@ class MainActivityTest {
         instantiateReachFiveClient("", CLIENT_ID)
     }
 
+    @Test
+    fun testSuccessfulClientConfigFetch() = clientTest(initialize = false) { client ->
+        val profile = aProfile()
+
+        client.signup(
+            profile,
+            success = { fail("This test should have failed because the `openid` scope should be missing prior to client initialization, causing auth token parsing to fail.") },
+            failure = { expectedError ->
+                // Signup success but no id_token returned
+                assertEquals(
+                    NO_ID_TOKEN,
+                    expectedError.message
+                )
+
+                client
+                    .initialize()
+                    .signup(
+                        profile,
+                        success = { fail("This test should have failed because the email is now already used.") },
+                        failure = { emailAlreadyExists ->
+                            assertEquals(
+                                "Bad Request",
+                                emailAlreadyExists.message
+                            )
+                        }
+                    )
+            }
+        )
+
+        sleep(1000)
```

Comment:
Is there a simple way to remove the sleep(x). It can impact the test execution determinism.

Reply:
I don't think so. We need to mock the callback and I've already lost a full day trying to do it. If you have a better idea, it would be great!

Reply:
Probably. I tried removing it to play around, and it just causes the tests to pass immediately since the callbacks aren't evaluated on time.

I think it's best to treat this problem separately in another PR.